part7/anecdotes/src/components/CreateNew.js

```diff
@@ -0,0 +1,67 @@
+import { useField } from '../hooks'
+// import { useRef } from 'react'
+
+const CreateNew = ({ addNew, history }) => {
+    // const input = useRef()
+    const content = useField('name')
+    const author = useField('author')
+    const info = useField('info')
+    const reset = useField('')
+
+
+    const handleSubmit = (e) => {
+        e.preventDefault()
+        addNew({
+            content: content.value,
+            author: author.value,
+            info: info.value,
+            votes: 0
+        })
+        history.push('/')
+    }
+
+    return (
+        <div>
+            <h2>create a new anecdote</h2>
+            <form >
+                {/* <form > */}
+                <div>
+                    content
+          <input
+                        {...content}
+                    // name={content.name}
+                    // value={content.value}
+                    // onChange={content.onChange}
+                    // {...content}
+                    // ref={input}
```

Comment:
Lielisks spreads, droši dzēs tās komentētās daļu ārā. 🎉

Reply:
> `hooks/index.js` **16. rinda**
> * Šis reseta spreads ir absolūti neloģisks.
> * Pastāsti, kāda bija tava doma? (neveicot korekcijas domā, vienkārši - kāds bija domu gājiens to spreadu te laižot, lai varu saprast kur knowledge gaps un izvest cauri)

to es arī nosaucu par partizānu gājienu. Es uzmetu acis [šim](https://reactjs.org/docs/jsx-in-depth.html#spread-attributes:~:text=You%20can%20also%20pick%20specific%20props,other%20props%20using%20the%20spread%20operator.) , un nodomāju: "pag, es nesaprotu, man te būs jāiedziļinās kārtīgāk. _reset_ man vajadzēs nošķirt, ja... un taad..." un atstāju to kā uzmetumu, un veicu tālākās darbības. atkal apskatījos uzmetumu un īsti vairs nespēju pateikt, kas tas, pie velna ir.
taču secināju, ka tīri tā, pavisam cieši samiedzot acis, viss strādā, bet es jūtu, ka tomēr kaut kas galīgi nav tā, tālab sūtīju augšā vērtēšanai, pirms es turpinu savas patvaļīgās _spread_ interpretācijas.